test/mainnet-fork/wstETH.t.sol

```diff
@@ -0,0 +1,283 @@
+        rateLimits.setRateLimitData(requestWithdrawKey, 1_000e18, uint256(1_000e18) / 1 days);
+        vm.stopPrank();
+
+        deal(Ethereum.WETH, address(almProxy), 1_000e18);
+
+        assertEq(rateLimits.getCurrentRateLimit(depositKey), 1_000e18);
+
+        assertEq(IERC20(Ethereum.WETH).balanceOf(address(almProxy)),   1_000e18);
+        assertEq(IERC20(Ethereum.WSTETH).balanceOf(address(almProxy)), 0);
+
+        vm.prank(relayer);
+        mainnetController.depositToWstETH(1_000e18);
+
+        assertEq(rateLimits.getCurrentRateLimit(depositKey),         0);
+        assertEq(rateLimits.getCurrentRateLimit(requestWithdrawKey), 1_000e18);
+
+        assertEq(IERC20(Ethereum.WETH).balanceOf(address(almProxy)),   0);
+        assertEq(IERC20(Ethereum.WSTETH).balanceOf(address(almProxy)), 823.029395390731625220e18);
+
+        vm.prank(relayer);
+        uint256[] memory requestIds = mainnetController.requestWithdrawFromWstETH(500e18);
+
+        assertEq(IERC20(Ethereum.WSTETH).balanceOf(address(almProxy)), 323.029395390731625220e18);
+
+        assertEq(rateLimits.getCurrentRateLimit(requestWithdrawKey), 500e18);
+
+        assertEq(requestIds.length, 1);
+
+        WithdrawalRequestStatus[] memory statuses = IWithdrawalQueue(Ethereum.unstETH).getWithdrawalStatus(requestIds);
+
+        assertApproxEqAbs(statuses[0].amountOfShares, 500e18, 1);
+
+        assertEq(statuses[0].amountOfStETH, 607.511715620589663161e18);
+        assertEq(statuses[0].owner,         address(almProxy));
+        assertEq(statuses[0].timestamp,     block.timestamp);
+        assertEq(statuses[0].isFinalized,   false);
+        assertEq(statuses[0].isClaimed,     false);
+    }
+
+}
+
+contract MainnetControllerClaimWithdrawalFromWstETHFailureTests is MainnetControllerWstETHTestBase {
+
+    function test_claimWithdrawalFromWstETH_notRelayer() external {
+        vm.expectRevert(abi.encodeWithSignature(
+            "AccessControlUnauthorizedAccount(address,bytes32)",
+            address(this),
+            RELAYER
+        ));
+        mainnetController.claimWithdrawalFromWstETH(1);
+    }
+
+}
+
+contract MainnetControllerClaimWithdrawalFromWstETHTests is MainnetControllerWstETHTestBase {
+
+    address finalizer = 0xae7ab96520DE3A18E5e111B5EaAb095312D7fE84;
+    
+    function test_claimWithdrawalFromWstETH() external {
+        bytes32 depositKey = RateLimitHelpers.makeAssetKey(
+            mainnetController.LIMIT_WSTETH_DEPOSIT(),
+            Ethereum.WETH
+        );
+        bytes32 requestWithdrawKey = RateLimitHelpers.makeAssetKey(
+            mainnetController.LIMIT_WSTETH_REQUEST_WITHDRAW(),
+            Ethereum.WSTETH
+        );
+
+        vm.startPrank(Ethereum.SPARK_PROXY);
+        rateLimits.setRateLimitData(depositKey,         10e18, uint256(10e18) / 1 days);
+        rateLimits.setRateLimitData(requestWithdrawKey, 10e18, uint256(10e18) / 1 days);
```

Comment:
These should use all the same larger numbers, just with a note that the smaller amount for withdrawal is necessary cause of market conditions (is it? can we do more?)